service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/JobProperties.java

```diff
@@ -126,6 +127,21 @@ public void setProgress(int progress) {
         this.progress = progress;
     }
 
+    /**
+     * @return System generated. Ignored at creation.
+     * If status == failure, this represents a string containing the reason.
+     */
+    public String getFailureReason() {
+        return failureReason;
+    }
+
+    /**
+     * @param failureReason the failure reason.
+     */
+    public void setFailureReason(String failureReason) {
+        this.failureReason = failureReason;
+    }
+
     /**
      * @return URI containing SAS token to a blob container that contains registry data to sync.
```

Comment:
"containing SAS token" is only true if the storageAuthenticationType is keyBased.

Check all these comments, and compare them against the updated ones in the c# PR.

Reply:
I looked at your PR and seeing same comment for this context. Did I miss something?

Reply:
Done